### PR TITLE
nemu: add support for static build of nemu

### PR DESCRIPTION
**Pull request by egernst**
Fixes: #401

Signed-off-by: Eric Ernst <eric.ernst@intel.com>

### CODE REVIEW

File: static-build/nemu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from ubuntu:16.04
+
+ARG NEMU_REPO
+# commit/tag/branch
+ARG NEMU_VERSION
+ARG NEMU_OVMF
+
+WORKDIR /root/nemu
+RUN apt-get update
```
Comment (devimc) [inline static-build/nemu/Dockerfile]: each `RUN` is a new layer in the final image, I wonder if won't be easier to use a script

File: static-build/nemu/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+RUN PREFIX=/opt/kata /root/configure-hypervisor.sh -s kata-nemu | xargs ./configure \
+       --with-pkgversion=kata-static
+
+RUN make -j$(nproc)
+RUN make install DESTDIR=/tmp/nemu-static
+
+RUN wget "${NEMU_OVMF}" && mv OVMF.fd /tmp/nemu-static/opt/kata/share/kata-nemu/
+RUN mv /tmp/nemu-static/opt/kata/bin/qemu-system-x86_64 /tmp/nemu-static/opt/kata/bin/nemu-system-x86_64
+
+RUN cd /tmp/nemu-static && tar -czvf kata-nemu-static.tar.gz *
```
Comment (devimc) [inline static-build/nemu/Dockerfile]: `tar -C /tmp/nemu-static -czvf kata-nemu-static.tar.gz`

File: static-build/nemu/Makefile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+MK_DIR :=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
```
Comment (devimc) [inline static-build/nemu/Makefile]: License

File: static-build/nemu/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,53 @@
+from ubuntu:16.04
```
[the remark this answers is not in the document]
Comment (egernst) [reply static-build/nemu/Dockerfile]: agreed.  as a starting point i'm just copying what is one with QEMU today.  Absolutely should have a better baseline (perhaps something more cutting edge like Clear, or something very minimal...).

File: static-build/nemu/Dockerfile (hunk 2 above)
Comment (egernst) [reply static-build/nemu/Dockerfile]: I actually need to chmod permissions on this too, so will need to add a step anyway.

Comment (egernst) [reply static-build/nemu/Dockerfile]: actually, fine as is (mixed it up with the virtiofsd part).

File: static-build/nemu/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	    libglib2.0-dev git \
+	    libltdl-dev \
+	    libpixman-1-dev \
+	    libtool \
+	    pkg-config \
+	    pkg-config \
+	    python \
+	    python-dev \
+	    rsync \
+	    wget \
```
Comment (jodh-intel) [inline static-build/nemu/Dockerfile]: We tend to use `curl` consistently across the project.

File: static-build/nemu/Dockerfile (hunk 1 above)
Comment (jodh-intel) [reply static-build/nemu/Dockerfile]: Agreed - I'd tend to try to keep the `Dockerfile` as small as possible and offload into a script.

**Pull request merged**